presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -77,7 +78,7 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
 
         Memo memo = new Memo(idAllocator, plan);
         Lookup lookup = Lookup.from(planNode -> Stream.of(memo.resolve(planNode)));
-        Matcher matcher = new PlanNodeMatcher(lookup);
+        Matcher matcher = new DefaultMatcher();
```

Comment:
Can you explain why this works and also put that in the commit message?

Reply:
PlanNodeMatcher was introduced in order to access Lookup when doing the
match. Now Lookup can be accessed in property accessor Function so there
is no need to have PlanNodeMatcher.

See `Patterns::source` and `Patterns::sources`.